packages/editor/src/AppModel/FieldModel.ts

```diff
@@ -1,18 +1,38 @@
-import { parseExpression } from '@sunmao-ui/shared';
+import { parseExpression, expChunkToString, SpecOptions } from '@sunmao-ui/shared';
 import * as acorn from 'acorn';
 import * as acornLoose from 'acorn-loose';
 import { simple as simpleWalk } from 'acorn-walk';
 import { flattenDeep, isArray, isObject } from 'lodash-es';
 import { isExpression } from '../validator/utils';
-import { ComponentId, IFieldModel, ModuleId } from './IAppModel';
+import {
+  ComponentId,
+  IAppModel,
+  IComponentModel,
+  ITraitModel,
+  IFieldModel,
+  ModuleId,
+  RefInfo,
+} from './IAppModel';
+import escodegen from 'escodegen';
+import { JSONSchema7 } from 'json-schema';
+
+type Flatten<Type> = Type extends Array<infer Item> ? Item : Type;
 
 export class FieldModel implements IFieldModel {
   isDynamic = false;
-  refs: Record<ComponentId | ModuleId, string[]> = {};
+  refs: Record<ComponentId | ModuleId, RefInfo> = {};
+  private nodes: Record<string, acorn.Node> = {};
```

Comment:
I think we could use more specific names. `refs` and `nodes` are too abstract here.

Reply:
How about the `componentRefs` and the `astNodes`?

Reply:
`astNodes` is good. `componentRefs` is still confusing. I think the problem is that `ref` is ambigious here.`RefInfo` have not only has `nodes`(seems like a combination of `astNode` and other), but also has `properties`(not sure what is it).